crates/proc_macro_api/src/process.rs

```diff
@@ -65,38 +52,15 @@ impl ProcMacroProcessSrv {
     where
         R: TryFrom<Response, Error = &'static str>,
     {
-        let (result_tx, result_rx) = bounded(0);
-        let sender = match self.inner.upgrade() {
-            None => return Err(tt::ExpansionError::Unknown("proc macro process is closed".into())),
-            Some(it) => it,
-        };
-        sender
-            .send(Task { req, result_tx })
-            .map_err(|_| tt::ExpansionError::Unknown("proc macro server crashed".into()))?;
-
-        let res = result_rx
-            .recv()
-            .map_err(|_| tt::ExpansionError::Unknown("proc macro server crashed".into()))?;
+        let mut guard = self.stdio.lock().unwrap_or_else(|e| e.into_inner());
+        let stdio = &mut *guard;
+        let (stdin, stdout) = (&mut stdio.0, &mut stdio.1);
 
-        match res {
-            Some(Response::Error(err)) => Err(tt::ExpansionError::ExpansionError(err.message)),
-            Some(res) => Ok(res.try_into().map_err(|err| {
-                tt::ExpansionError::Unknown(format!("Fail to get response, reason : {:#?} ", err))
-            })?),
-            None => Err(tt::ExpansionError::Unknown("Empty result".into())),
-        }
-    }
-}
-
-fn client_loop(task_rx: Receiver<Task>, mut process: Process) {
-    let (mut stdin, mut stdout) = process.stdio().expect("couldn't access child stdio");
-
-    let mut buf = String::new();
-
-    for Task { req, result_tx } in task_rx {
-        match send_request(&mut stdin, &mut stdout, req, &mut buf) {
-            Ok(res) => result_tx.send(res).unwrap(),
+        let mut buf = String::new();
+        let res = match send_request(stdin, stdout, req, &mut buf) {
+            Ok(res) => res,
             Err(err) => {
+                let mut process = self.process.lock().unwrap_or_else(|e| e.into_inner());
                 log::error!(
                     "proc macro server crashed, server process state: {:?}, server request error: {:?}",
                     process.child.try_wait(),
```

Comment:
while we are at it, let's move `try_wait` out of potentially disabled logging :)